SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -88,19 +89,25 @@ - (void)sdl_stopEventListening {
 #pragma mark - EAAccessory Notifications
 
 - (void)sdl_accessoryConnected:(NSNotification *)notification {
+    EAAccessory *accessory = [notification.userInfo objectForKey:EAAccessoryKey];
     NSMutableString *logMessage = [NSMutableString stringWithFormat:@"Accessory Connected, Opening in %0.03fs", self.retryDelay];
     [SDLDebugTool logInfo:logMessage withType:SDLDebugType_Transport_iAP toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
 
     self.retryCounter = 0;
-    [self performSelector:@selector(connect) withObject:nil afterDelay:self.retryDelay];
+
+    [self performSelector:@selector(sdl_connect:) withObject:accessory afterDelay:self.retryDelay];
 }
 
 - (void)sdl_accessoryDisconnected:(NSNotification *)notification {
     [SDLDebugTool logInfo:@"Accessory Disconnected Event" withType:SDLDebugType_Transport_iAP toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
+    [self sdl_stopEventListening];
 
     // Only check for the data session, the control session is handled separately
     EAAccessory *accessory = [notification.userInfo objectForKey:EAAccessoryKey];
-    if (accessory.connectionID == self.session.accessory.connectionID) {
+    if (accessory.connectionID != self.session.accessory.connectionID) {
+         [SDLDebugTool logInfo:@"Accessory connection ID mismatch!!!" withType:SDLDebugType_Transport_iAP toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
+    }
```

Comment:
This change was made because, due to zombie object references, the EAAccessory framework was sending disconnect notifications with a different connection ID to the transport. If the disconnect sequence is not executed, these objects hang around and cause crashes shortly thereafter. I recommend based on our testing that we disconnect even if the connection ID doesn't match.